server/src/main/java/org/elasticsearch/script/field/LongDocValuesField.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.util.ArrayUtil;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class LongDocValuesField implements DocValuesField<Long>, ScriptDocValues.Supplier<Long> {
+
+    protected final SortedNumericDocValues input;
+    protected final String name;
+
+    protected long[] values = new long[0];
+    protected int count;
+
+    private ScriptDocValues.Longs longs = null;
+
+    public LongDocValuesField(SortedNumericDocValues input, String name) {
+        this.input = input;
+        this.name = name;
+    }
+
+    /**
+     * Set the current document ID.
+     *
+     * @param docId
```

Comment:
Please delete this line since docID doesn't really need a comment to be clear.

Reply:
Removed.